eva-server/src/main/java/uk/ac/ebi/eva/server/repository/VariantEntityRepositoryImpl.java

```diff
@@ -38,33 +41,110 @@
  * to provide an explicit implementation of the region query, using a margin for efficiency.
  */
+@Component
 public class VariantEntityRepositoryImpl implements VariantEntityRepositoryCustom {
 
-    private MongoDbFactory mongoDbFactory;
     private MongoTemplate mongoTemplate;
-    private MappingMongoConverter mappingMongoConverter;
 
     private final int MARGIN = 1000000;
 
     @Autowired
     public VariantEntityRepositoryImpl(MongoDbFactory mongoDbFactory, MappingMongoConverter mappingMongoConverter) {
-        this.mongoDbFactory = mongoDbFactory;
-        this.mappingMongoConverter = mappingMongoConverter;
         mongoTemplate = new MongoTemplate(mongoDbFactory, mappingMongoConverter);
     }
 
-    public List<VariantEntity> findByChrAndStartWithMarginAndEndWithMargin(String chr, int start, int end) {
+    @Override
+    public List<VariantEntity> findByRegionAndComplexFilters(String chr, int start, int end,
+                                                             List<String> studies, List<String> consequenceType,
+                                                             VariantEntityRepository.RelationalOperator mafOperator,
+                                                             Double mafValue,
+                                                             VariantEntityRepository.RelationalOperator polyphenScoreOperator,
+                                                             Double polyphenScoreValue,
+                                                             VariantEntityRepository.RelationalOperator siftOperator,
+                                                             Double siftValue, Pageable pageable) {
         Query query = new Query(Criteria
                                         .where("chr").is(chr)
                                         .and("start").lte(end).gt(start - MARGIN)
                                         .and("end").gte(start).lt(end + MARGIN)
         );
 
+        if (consequenceType != null && !consequenceType.isEmpty()) {
+            queryConsequenceType(query, consequenceType);
+        }
+
+        if (mafValue != null && mafOperator != VariantEntityRepository.RelationalOperator.NONE) {
+            queryMaf(query, mafValue, mafOperator);
+        }
+
+        if (polyphenScoreValue != null && polyphenScoreOperator != VariantEntityRepository.RelationalOperator.NONE) {
+            queryPolyphenScore(query, polyphenScoreValue, polyphenScoreOperator);
+        }
+
+        if (siftValue != null && siftOperator != VariantEntityRepository.RelationalOperator.NONE) {
+            querySift(query, siftValue, siftOperator);
+        }
+
+        if (studies != null && !studies.isEmpty()) {
+            queryStudies(query, studies);
+        }
+
         ArrayList<String> sortProps = new ArrayList<String>();
         sortProps.add("chr");
         sortProps.add("start");
         query.with(new Sort(Sort.Direction.ASC, sortProps));
 
+        query.with(pageable);
```

Comment:
If `pageable` is null we must enforce some default value, like a first page with 10 elements. Otherwise queries could still return all the potential values.